docs/aws/audit/route53monitoring/rules/domain_expiry_30_days.mdx

```diff
@@ -23,6 +23,270 @@ CBP
+
+2. List hosted zones: Use the following command to list all the hosted zones in your AWS account:
+
+   ```
+   aws route53 list-hosted-zones
+   ```
+   This command will return a list of all the hosted zones in your AWS account, including their ID, name, and other information.
+
+3. List resource record sets: For each hosted zone, use the following command to list all the resource record sets:
+
+   ```
+   aws route53 list-resource-record-sets --hosted-zone-id <hosted-zone-id>
+   ```
+   Replace `<hosted-zone-id>` with the ID of the hosted zone you want to check. This command will return a list of all the resource record sets in the specified hosted zone, including their name, type, and other information.
+
+4. Check domain expiry: For each resource record set of type `SOA` (Start of Authority), check the `RDATA` field for the expiry value. If the expiry value is less than or equal to 30 days, then the domain is set to expire in 30 days or less. Note that the expiry value is specified in seconds, so you need to convert it to days by dividing by 86400 (the number of seconds in a day).
+
+   Unfortunately, AWS CLI does not provide a direct way to check the expiry date of a domain. You may need to use other tools or services, or check the domain expiry manually through the AWS Management Console.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+```python
+pip install boto3
+```
+
+2. Import the necessary libraries and initialize the Route53 client. You will need to provide your AWS credentials (access key and secret key) which you can get from your AWS console.
+```python
+import boto3
+from datetime import datetime, timedelta
+
+# Initialize the Route53 client
+route53 = boto3.client('route53', aws_access_key_id='YOUR_ACCESS_KEY', aws_secret_access_key='YOUR_SECRET_KEY')
+```
+
+3. Fetch the list of domain names and their expiry dates. The 'list_domains' function of the Route53 client can be used to get the list of domain names. For each domain name, you can get the expiry date using the 'get_domain_detail' function.
+```python
+# Get the list of domain names
+domains = route53.list_domains()
+
+for domain in domains['Domains']:
+    # Get the domain details
+    domain_details = route53.get_domain_detail(DomainName=domain['DomainName'])
+    
+    # Get the expiry date
+    expiry_date = domain_details['Expiry']
+```
+
+4. Check if the domain is expiring in the next 30 days. You can do this by comparing the expiry date with the current date plus 30 days. If the expiry date is earlier, then the domain is expiring in the next 30 days.
+```python
+# Get the current date plus 30 days
+current_date_plus_30 = datetime.now() + timedelta(days=30)
+
+# Check if the domain is expiring in the next 30 days
+if expiry_date <= current_date_plus_30:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>